pkg/advisory/validate.go

```diff
@@ -266,11 +182,6 @@ func (opts ValidateOptions) validateBuildConfigurationExistence(pkgName string)
 	}
 
 	if _, ok := opts.distroPackageMap[pkgName]; !ok {
-		// Double check if the package is in index and move on
-		if _, ok := opts.apkIndexPackageMap[pkgName]; ok {
-			// the package build configuration has moved out but not removed from the index or apks
-			return nil
-		}
```

Comment:
@ajayk Hi! I just wanted to get you to sanity check me here, since this is undoing a few lines of your work. The logic removed here was allowing the advisory data to refer to a package that's not defined as a Melange file in the distro, as long as it's listed in the APKINDEX.

But `validateBuildConfigurationExistence` is only called for **net new documents** — so the data that's being added _right now_ by the user.

You had a explanation comment here that I'm also removing:

```go
// This would change from March 13,2024 as package yamls would be deleted for EOL and non latest versions
// Only the package (apk) versions would exists in wolfi and wolfi index but no build config
```

If we're _deleting_ documents, I don't believe this logic actually comes into play at all. And I think we do want to make sure that documents being added in a PR **_are_** referencing a package that's still in the distro.

Curious for your thoughts here!